src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1541,6 +1543,60 @@ def reoffset(o):
 
         minimize_int(offset, reoffset)
 
+    def shrink_offset_pairs(self):
+        """Lower any two blocks offset from each other the same ammount.
+
+        Try lowering every non-zero pair of non-shrinking blocks by a
+        common offset.
+
+        """
+        self.debug('Shrinking offset pairs.')
+
+        current = self.shrink_target
+        blocked = [current.buffer[u:v] for u, v in current.blocks]
+
+        n = len(blocked)
+
+        shrink_blocks = [b for b in range(n) if self.is_shrinking_block(b)]
```

Comment:
Use `hrange` instead of `range`, for Py2/Py3 compatible iterator behaviour (from hypothesis.internal.compat; already imported)